### PR TITLE
NR-38551 -- Unit Tests

### PR DESCRIPTION
**Pull request by CalvinAllen**
Migrates unit tests to xunit and sets up CI server to run tests and provide code coverage.

Also includes the manifest change to require 17.2 minimum version for 2022

### CODE REVIEW

File: vs/docs/dev.md [hunk 1]
```diff
@@ -91,38 +91,29 @@ To debug the CodeStream LSP agent you will need both Visual Studio and VS Code.
...
-```shell
-cd build
-.\Pre-Build.ps1
-.\Bump-Version.ps1 -BuildNumber $env:build_number  -BumpBuild
-.\Build.ps1
-```
+Using 'Debug' and 'x86' will produce a Visual Studio 2019 compatible artifact at `.\vs\src\CodeStream.VisualStudio.Vxix.x86/bin/x86/Debug/codestream-vs.vsix` and can be debugged directly using the Visual Studio 2019 experimental instance.
 
-The build portion of the artifact version can be updated using `.\Bump-Version.ps1 -BuildNumber 666 -BumpBuild`. this will convert `0.1.0` to `0.1.0.666`
+Using 'Debug' and 'x64' will procuce a Visual Studio 2022 compatible artifact at `.\vs\src\CodeStream.VisualStudio.Vxix.x64/bin/x64/Debug/codestream-vs-22.vsix` and can be debugged directly using the Visual Studio 2022 experimental instance.
```
Comment (bcanzanella) [inline vs/docs/dev.md:100]: procuce typo :/

File: vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs [hunk 2]
```diff
@@ -5,21 +5,19 @@
...
+
+using Xunit;
 
 namespace CodeStream.VisualStudio.UnitTests {
 	//https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ff576068%28v%3dvs.100%29
 	//https://stackoverflow.com/questions/8324302/how-to-debug-mef-exception
 
-
-	[TestClass]
 	public class CompositionTests {
```
Comment (bcanzanella) [inline vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs:15]: wonder if these "tests" would even be useful...

**Comment by bcanzanella** [comment]
🔥

**Review by bcanzanella — APPROVED** [review]

File: vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs (hunk 2 above)
Comment (CalvinAllen) [reply vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs:15]: Fixed

File: vs/docs/dev.md (hunk 1 above)
Comment (CalvinAllen) [reply vs/docs/dev.md]: Fixed

File: vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs (hunk 2 above)
Comment (CalvinAllen) [reply vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs:15]: I think theres still a lot to do on the tests, overall, now that they actually run and produce results / reports that we can leverage.

Comment (bcanzanella) [reply vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs:15]: sorry, i mean't CompositionTests.cs specifically (not sure what i was doing with that...)

Comment (CalvinAllen) [reply vs/src/CodeStream.VisualStudio.UnitTests/CompositionTests.cs:15]: I think this was the one trying to validate the MEF graph using ApprovalTests

**Pull request merged**